src/lib/update-lock.ts

```diff
@@ -75,13 +64,52 @@ const locksTaken: { [lockName: string]: boolean } = {};
 process.on('exit', () => {
 	for (const lockName of _.keys(locksTaken)) {
 		try {
-			lockFile.unlockSync(lockName);
+			lockfile.unlockSync(lockName);
 		} catch (e) {
 			// Ignore unlocking errors
 		}
 	}
 });
 
+/**
+ * Cleans up any Supervisor-created lockfiles, intended to be called
+ * on process startup to remove any lockfiles from previous Supervisor process.
+ *
+ * Supervisor-created lockfiles are distinguished by their UID of 65534,
+ * corresponding to the `nobody` user.
+ */
+export async function cleanupLocks() {
+	// Should be /mnt/root/tmp/balena-supervisor/services
+	const lockfileParentDir = getPathOnHost(BASE_LOCK_DIR);
+
+	try {
+		// Unix `find` command for listing all Supervisor-created lockfiles
+		const findCommand = `find ${lockfileParentDir} -type f -user ${lockfile.NOBODY_UID} -name "*updates.lock"`;
+
+		// Find paths for leftover Supervisor lockfiles
+		const { stdout, stderr } = await exec(findCommand);
```

Comment:
Using Unix `find` is more efficient than implementing a recursive directory search in TypeScript